client/request.go

```diff
@@ -114,13 +114,31 @@ func (cli *Client) buildRequest(method, path string, body io.Reader, headers hea
 }
 
 func (cli *Client) sendRequest(ctx context.Context, method, path string, query url.Values, body io.Reader, headers headers) (serverResponse, error) {
-	req, err := cli.buildRequest(method, cli.getAPIPath(path, query), body, headers)
+	requestURL := cli.getAPIPath(path, query)
+
+	if err := cli.validatePath(requestURL); err != nil {
+		return serverResponse{}, err
+	}
+	req, err := cli.buildRequest(method, requestURL, body, headers)
 	if err != nil {
 		return serverResponse{}, err
 	}
 	return cli.doRequest(ctx, req)
 }
 
+func (cli *Client) validatePath(url string) error {
+	forbiddenStrings := []string{
+		"../", // causes strange server redirect 301 replies
```

Comment:
I'm really not sure about doing validations here.
It puts new restrictions on things that were previously not restricted and makes assumptions about what the daemon can and cannot accept.

I think it would just be better to handle the redirect gracefully.
Open to what others think, though.